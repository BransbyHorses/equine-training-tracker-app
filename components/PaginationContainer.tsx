import React, { ReactNode, useState } from "react";
import { Box } from "@mui/material";
import Pagination from "@mui/material/Pagination";

const PaginationContainer = ({
	children,
	count,
}: {
	children: ReactNode;
	count: number;
}) => {
	const [page, setPage] = useState(1);
	const resultsPerpage = 10;
	const resultsFrom = (page - 1) * resultsPerpage + 1;
	const resultsTo = page * resultsPerpage;

	const handleNavigationChange = (event: any, value: any) => {
		setPage(parseInt(value));
	};

	return (
		<>
			{React.Children.map(children, (child, index) => {
				if (index + 1 >= resultsFrom && index + 1 <= resultsTo) return child;
			})}
			<Box sx={{ display: "flex", justifyContent: "flex-end" }}>
				<Pagination
					sx={{ mt: 3 }}
					page={page}
					count={Math.ceil(count / 10)}
					onChange={handleNavigationChange}
				/>
			</Box>
		</>
	);
};

export default PaginationContainer;
