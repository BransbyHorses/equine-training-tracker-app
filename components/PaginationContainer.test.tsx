import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import PaginationContainer from "./PaginationContainer";

describe("PaginationContainer", () => {
	it("renders with all props", () => {
		const allProps = {
			count: 0,
		};
		const tree = renderer.create(
			<PaginationContainer {...allProps}>
				{[
					1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
				].map((c) => {
					return (
						<div key={c} id={`${c.toString()}`}>
							Content {c}
						</div>
					);
				})}
			</PaginationContainer>
		);
		expect(tree).toMatchSnapshot();
	});

	it("displays 10 items per page", () => {
		render(
			<PaginationContainer count={20}>
				{[
					1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
				].map((c) => {
					return (
						<div key={c} id={`${c.toString()}`}>
							Content {c}
						</div>
					);
				})}
			</PaginationContainer>
		);
		expect(screen.getByText("Content 1")).toBeInTheDocument();
		expect(screen.getByText("Content 2")).toBeInTheDocument();
		expect(screen.getByText("Content 3")).toBeInTheDocument();
		expect(screen.getByText("Content 4")).toBeInTheDocument();
		expect(screen.getByText("Content 5")).toBeInTheDocument();
		expect(screen.getByText("Content 6")).toBeInTheDocument();
		expect(screen.getByText("Content 7")).toBeInTheDocument();
		expect(screen.getByText("Content 8")).toBeInTheDocument();
		expect(screen.getByText("Content 9")).toBeInTheDocument();
		expect(screen.getByText("Content 10")).toBeInTheDocument();
		expect(screen.queryByText("Content 11")).toBeNull();
		expect(screen.queryByText("Content 20")).toBeNull();
	});

	it("pagination menu not displayed when children length is 0", () => {
		const tree = renderer.create(
			<PaginationContainer count={0}>
				{[].map(() => {
					return <></>;
				})}
			</PaginationContainer>
		);
		expect(tree).toMatchSnapshot();
	});
});
