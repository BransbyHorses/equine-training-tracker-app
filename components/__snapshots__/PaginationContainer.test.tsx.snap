// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationContainer pagination menu not displayed when children length is 0 1`] = `
<div
  className="MuiBox-root css-13w632d"
>
  <nav
    aria-label="pagination navigation"
    className="MuiPagination-root MuiPagination-text css-pg6uc-MuiPagination-root"
  >
    <ul
      className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`PaginationContainer renders with all props 1`] = `
[
  <div
    id="1"
  >
    Content 
    1
  </div>,
  <div
    id="2"
  >
    Content 
    2
  </div>,
  <div
    id="3"
  >
    Content 
    3
  </div>,
  <div
    id="4"
  >
    Content 
    4
  </div>,
  <div
    id="5"
  >
    Content 
    5
  </div>,
  <div
    id="6"
  >
    Content 
    6
  </div>,
  <div
    id="7"
  >
    Content 
    7
  </div>,
  <div
    id="8"
  >
    Content 
    8
  </div>,
  <div
    id="9"
  >
    Content 
    9
  </div>,
  <div
    id="10"
  >
    Content 
    10
  </div>,
  <div
    className="MuiBox-root css-13w632d"
  >
    <nav
      aria-label="pagination navigation"
      className="MuiPagination-root MuiPagination-text css-pg6uc-MuiPagination-root"
    >
      <ul
        className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>,
]
`;
